lib/src/components/NxCollapsibleItemsSelect/NxCollapsibleRadioSelect/__tests__/NxCollapsibleRadioSelect.test.tsx

```diff
@@ -4,173 +4,495 @@
+    });
+
+    describe('onFilterChange prop', function() {
+      it('calls onFilterChange when filter value changes', async function() {
+        const user = userEvent.setup(),
+            onFilterChange = jest.fn().mockImplementation((_, evt) => { evt.persist(); }),
+            inputEl = quickFilterRender({ onFilterChange, filter: ''}).getByRole('textbox');
+
+        expect(onFilterChange).not.toHaveBeenCalled();
+
+        inputEl.focus();
+        await user.keyboard('a');
+
+        expect(onFilterChange).toHaveBeenCalledWith('a', expect.objectContaining({ target: inputEl }));
+      });
+
+      it('fires onFilterChange with the empty string when the Escape key is pressed', async function() {
+        const user = userEvent.setup(),
+            onFilterChange = jest.fn(),
+            el = quickFilterRender({ filter: 'b', onFilterChange }),
+            inputEl = el.getByRole('textbox');
+
+        expect(onFilterChange).not.toHaveBeenCalled();
+
+        inputEl.focus();
+        await user.keyboard('[Escape]');
+
+        expect(onFilterChange).toHaveBeenCalledWith('');
+      });
+
+      it('fires onFilterChange with empty string when clear filter button is clicked', async function() {
+        const user = userEvent.setup(),
+            onFilterChange = jest.fn(),
+            view = quickFilterRender({ filter: 'b', onFilterChange });
+
+        await runTimers();
+        const clearBtn = view.getByRole('button', { name: 'Clear filter' });
+
+        expect(onFilterChange).not.toHaveBeenCalled();
+
+        await user.click(clearBtn);
+
+        expect(onFilterChange).toHaveBeenCalledWith('');
+      });
+    });
+
+    it('calls preventDefault when the ESC is pressed and filter value is not empty', async function() {
+      const inputEl = quickFilterRender({ filter: 'b' }).getByRole('textbox'),
+          keyEvent = createEvent.keyDown(inputEl, { key: 'Escape' });
+
+      inputEl.focus();
+      fireEvent(inputEl, keyEvent);
+
+      expect(keyEvent.defaultPrevented).toBe(true);
     });
 
-    it('renders <Counter> with selected option name even if the option id is null', function () {
-      const renderPropWrapper = getShallow({selectedId: null}).renderProp('renderCounter')();
-      expect(renderPropWrapper).toContainReact(<Counter>Null</Counter>);
+    it('doesn\'t call preventDefault when the ESC is pressed and filter value is empty', async function() {
+      const inputEl = quickFilterRender().getByRole('textbox'),
+          keyEvent = createEvent.keyDown(inputEl, { key: 'Escape' });
+
+      inputEl.focus();
+      fireEvent(inputEl, keyEvent);
+
+      expect(keyEvent.defaultPrevented).toBe(false);
+    });
+
+    describe('when options are filtered', function() {
+      it('renders all options if filteredOptions prop is not provided', function() {
+        const view = quickFilterRender({ filteredOptions: null }),
+            options = view.getByRole('menu').children;
```

Comment:
Probably better to avoid using `.children` and to assert based on the checkboxes a11y names instead

Reply:
> Probably better to avoid using `.children` and to assert based on the checkboxes a11y names instead

Makes sense. Added in both tests here: [5f28839](https://github.com/sonatype/sonatype-react-shared-components/pull/932/commits/5f28839ddb0adadacfbe8cc2e3d0207b2221cab3)